pandas/core/arrays/categorical.py

```diff
@@ -2611,6 +2611,11 @@ def _get_codes_for_values(values, categories):
         values = ensure_object(values)
         categories = ensure_object(categories)
 
+    if isinstance(categories, ABCIndexClass):
```

Comment:
hmm shouldn't this be in the *caller* to this routine?

Reply:
I think it's best here. We call it in ~3 places, and I think each would have to repeat this check & the object casting stuff.